lib/mix/lib/mix/scm/git.ex

```diff
@@ -25,17 +25,24 @@ defmodule Mix.SCM.Git do
   end
 
   def accepts_options(_app, opts) do
-    cond do
-      gh = opts[:github] ->
-        opts
-        |> Keyword.delete(:github)
-        |> Keyword.put(:git, "https://github.com/#{gh}.git")
-        |> validate_git_options
-      opts[:git] ->
-        opts
-        |> validate_git_options
-      true ->
-        nil
+    opts =
+      cond do
+        gh = opts[:github] ->
+          opts
+          |> Keyword.delete(:github)
+          |> Keyword.put(:git, "https://github.com/#{gh}.git")
+          |> validate_git_options
+        opts[:git] ->
+          opts
+          |> validate_git_options
+        true ->
+          nil
+      end
+    if opts[:sparse] do
```

Comment:
You can store the `:dest` option as `:checkout` or `:git_dest` so you don't need to use replace_suffix later on.